From 6f1ecba7774293d3db34226b09339b32a28f24a3 Mon Sep 17 00:00:00 2001
From: jayeshkrishna <jayesh@mcs.anl.gov>
Date: Wed, 13 Apr 2022 13:40:54 -0500
Subject: [PATCH] Build old perf tool only if gptl is available

The older version the performance tool requires GPTL. So making
sure that its built only if GPTL is available (PIO_ENABLE_TIMING
is ON)
---
 tests/performance/CMakeLists.txt | 27 ++++++++++++++++++---------
 1 file changed, 18 insertions(+), 9 deletions(-)

diff --git a/tests/performance/CMakeLists.txt b/tests/performance/CMakeLists.txt
index f1c3bd6937..094c50e79f 100644
--- a/tests/performance/CMakeLists.txt
+++ b/tests/performance/CMakeLists.txt
@@ -16,11 +16,14 @@ endif ()
 #==============================================================================
 #  DEFINE THE TARGETS
 #==============================================================================
-add_executable (pioperf EXCLUDE_FROM_ALL
-  pioperformance.F90)
-set_property(TARGET pioperf PROPERTY LINKER_LANGUAGE ${PIO_LINKER_LANGUAGE})
-target_link_libraries (pioperf piof)
-add_dependencies (tests pioperf)
+if (PIO_ENABLE_TIMING)
+  # This tool requires GPTL
+  add_executable (pioperf EXCLUDE_FROM_ALL
+    pioperformance.F90)
+  set_property(TARGET pioperf PROPERTY LINKER_LANGUAGE ${PIO_LINKER_LANGUAGE})
+  target_link_libraries (pioperf piof)
+  add_dependencies (tests pioperf)
+endif ()
 
 add_executable (pioperf_rearr EXCLUDE_FROM_ALL
   pioperformance_rearr.F90)
@@ -39,8 +42,11 @@ endif ()
 
 # Add compiler-specifc options for the tools
 if ("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "GNU")
-  target_compile_options (pioperf
-    PRIVATE -ffree-line-length-none)
+  if (PIO_ENABLE_TIMING)
+    # This tool requires GPTL
+    target_compile_options (pioperf
+      PRIVATE -ffree-line-length-none)
+  endif ()
   target_compile_options (pioperf_rearr
     PRIVATE -ffree-line-length-none)
   if (CMAKE_Fortran_COMPILER_VERSION VERSION_GREATER_EQUAL "10.0")
@@ -49,8 +55,11 @@ if ("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "GNU")
 endif()
 
 if (PIO_HDF5_LOGGING)
-  target_compile_definitions (pioperf
-    PUBLIC LOGGING)
+  if (PIO_ENABLE_TIMING)
+    # This tool requires GPTL
+    target_compile_definitions (pioperf
+      PUBLIC LOGGING)
+  endif ()
   target_compile_definitions (pioperf_rearr
     PUBLIC LOGGING)
 endif ()
